build-heads.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+build_only="${1}"
+implementations=$(find ./implementations -maxdepth 1 -name "*")
```

Comment:
`./implementations` can be just `implementations`. Probably good to add `-type d`.